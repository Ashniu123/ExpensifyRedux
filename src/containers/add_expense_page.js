import React, { Component } from 'react';
import { connect } from 'react-redux';

import ExpenseForm from '../components/expense_form';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends Component {
  constructor(props){
    super(props);

    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(expense) {
    this.props.startAddExpense(expense);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Add Expense</h1>
            </div>
        </div>
        <div className="content-container">
            <ExpenseForm onSubmit={this.handleSubmit} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
