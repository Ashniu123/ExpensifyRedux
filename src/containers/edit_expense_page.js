import React, { Component } from 'react';
import { connect } from 'react-redux';

import ExpenseForm from '../components/expense_form';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(expense) {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }

  handleClick(e) {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
              <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm expense={this.props.expense} onSubmit={this.handleSubmit} />
          <button className="button button__secondary" onClick={this.handleClick}>Remove Expense</button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, updates) => dispatch(startEditExpense(id, updates)),
  startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
