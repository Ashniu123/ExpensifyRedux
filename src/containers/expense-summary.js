import React, { Component } from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export class ExpenseSummary extends Component {
    render() {
        const expensesWord = this.props.expenseCount === 1 ? 'expense' : 'expenses';
        const formattedTotalExpense = numeral(this.props.expenseTotal/100).format('$0,0.00');

        return (
            <div className="page-header">
                <div className="content-container">
                    <h3 className="page-header__title">
                        Viewing <strong>{this.props.expenseCount}</strong> {expensesWord} totalling <strong>{formattedTotalExpense}</strong>
                    </h3>
                    <div className="page-header__actions">
                        <Link className="button" to="/create">Add Expense</Link>
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filter);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: selectExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpenseSummary);