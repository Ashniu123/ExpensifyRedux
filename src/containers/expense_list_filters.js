import React, { Component } from 'react';
import { connect } from 'react-redux';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends Component {
    constructor(props) {
        super(props);

        this.state = {
            calendarFocused: null
        };

        this.handleSortByChange = this.handleSortByChange.bind(this);
        this.handleFocusChange = this.handleFocusChange.bind(this);
        this.handleDatesChange = this.handleDatesChange.bind(this);
        this.handleTextFilterChange = this.handleTextFilterChange.bind(this);
    }

    handleSortByChange(e) {
        switch (e.target.value) {
            case "date": this.props.sortByDate();
                break;
            case "amount": this.props.sortByAmount();
                break;
            default: break;
        }
    }

    handleDatesChange({ startDate, endDate }) {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    }

    handleFocusChange(focused) {
        this.setState({ calendarFocused: focused });
    }

    handleTextFilterChange(event) {
        this.props.setTextFilter(event.target.value);
    }

    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input className="text-input" type="text" placeholder="Search Expenses..." value={this.props.filter.text} onChange={this.handleTextFilterChange} />
                    </div>
                    <div className="input-group__item">
                        <select className="select" value={this.props.filter.sortBy} onChange={this.handleSortByChange}>
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filter.startDate}
                            endDate={this.props.filter.endDate}
                            onDatesChange={this.handleDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.handleFocusChange}
                            numberOfMonths={1} isOutsideRange={(day) => false}
                            showClearDates={true}
                        />
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    // console.log(state);
    return {
        filter: state.filter
    };
};

const mapDispatchToProps = (dispatch) => ({
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: (startDate) => dispatch(setStartDate(startDate)),
    setEndDate: (endDate) => dispatch(setEndDate(endDate)),
    setTextFilter: (text) => dispatch(setTextFilter(text))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)

