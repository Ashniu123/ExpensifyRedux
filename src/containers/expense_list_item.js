import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <Link className="list-item" to={`/edit/${id}`}>
        <div >
            <h4 className="list-item__title">{description}</h4>
            <span className="list-item__sub-title">{moment(createdAt).format('MMMM Do, YYYY')}</span>
        </div>
        <h4 className="list-item__data">{numeral(amount / 100).format('$0,0.00')}</h4>
    </Link>
);

const mapStateToProps = (state) => {
    return {
        expenses: state.expenses
    };
};

export default connect(mapStateToProps)(ExpenseListItem);