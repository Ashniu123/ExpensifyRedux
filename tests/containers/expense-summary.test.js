import React from 'react';
import { shallow } from 'enzyme';

import { ExpenseSummary } from '../../src/containers/expense-summary';

test('should render ExpenseSummary with single expense', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expenseTotal={100} />)
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenseCount={10} expenseTotal={2022230} />)
    expect(wrapper).toMatchSnapshot();
});