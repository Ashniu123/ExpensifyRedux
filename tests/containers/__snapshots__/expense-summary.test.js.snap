// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h3
      className="page-header__title"
    >
      Viewing 
      <strong>
        10
      </strong>
       
      expenses
       totalling 
      <strong>
        $20,222.30
      </strong>
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary with single expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h3
      className="page-header__title"
    >
      Viewing 
      <strong>
        1
      </strong>
       
      expense
       totalling 
      <strong>
        $1.00
      </strong>
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
